src/cli/migrate/nodeVersion/packageJsonChecks.ts

```diff
@@ -0,0 +1,75 @@
+import findUp from 'find-up';
+import fs from 'fs-extra';
+
+import { log } from '../../../utils/logging';
+
+const getParentPackageJson = async () => {
+  const packageJsonPath = await findUp('package.json', { cwd: process.cwd() });
+  if (!packageJsonPath) {
+    throw new Error('package.json not found');
+  }
+  return fs.readFile(packageJsonPath);
+};
+
+const isTypeError = (error: unknown): error is TypeError =>
+  error instanceof TypeError &&
+  error.message.includes('Cannot read properties of undefined');
+
+const isSyntaxError = (error: unknown): error is SyntaxError =>
+  error instanceof SyntaxError && error.message.includes('Unexpected token');
+
+export const validServerlessVersion = async (): Promise<boolean> => {
+  const packageJson = await getParentPackageJson();
+
+  try {
+    const serverlessVersion = (
+      JSON.parse(packageJson.toString()) as {
+        devDependencies: Record<string, string>;
+      }
+    ).devDependencies.serverless;
+    if (!serverlessVersion) {
+      return true;
+    }
+
+    if (!serverlessVersion.startsWith('4')) {
+      log.warn(
+        'Serverless version not supported, please upgrade to 4.x to automatically update serverless files',
+      );
+      return false;
+    }
+  } catch (error) {
+    if (isTypeError(error) || isSyntaxError(error)) {
+      return true;
+    }
+    throw error;
+  }
```

Comment:
An alternative here may be to use Zod (you'll still need to consider whether we need to handle errors from `JSON.parse` though):

```typescript
const result = 
  z.object({
    devDependencies: z.object({
      serverless: z.string(),
    }),
  }).transform((data) => data.devDependencies.serverless).safeParse(JSON.parse(packageJson));
  
if (!result.success) {
  return true;
}

const serverlessVersion = result.data;
```